### ISSUE TITLE
JDK 17, Windows: Records annotations 

### ISSUE THREAD

**Issue by Karm** [graalvm/mandrel#370]
After merging a test https://github.com/Karm/mandrel-integration-tests/pull/79
it seems https://github.com/graalvm/mandrel/pull/360 does not work as expected on Windows.

```
2022-03-24T15:09:36.5613162Z [1/7] Initializing...                                                                                    (9.9s @ 0.12GB)
2022-03-24T15:09:36.5698464Z  Version info: 'GraalVM 22.0.0.2-Final Java 17 Mandrel Distribution'
2022-03-24T15:10:14.1773628Z [2/7] Performing analysis...  [*******]                                                                 (37.3s @ 0.89GB)
2022-03-24T15:10:14.1849038Z    2,938 (75.86%) of  3,873 classes reachable
2022-03-24T15:10:14.1849783Z    3,607 (59.72%) of  6,040 fields reachable
2022-03-24T15:10:14.1961145Z   14,036 (45.40%) of 30,918 methods reachable
2022-03-24T15:10:14.1965965Z       29 classes,    33 fields, and   296 methods registered for reflection
2022-03-24T15:10:14.1966917Z       61 classes,    53 fields, and    51 methods registered for JNI access
2022-03-24T15:10:14.2122833Z 
2022-03-24T15:10:14.2124080Z Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported method java.lang.Class.getConstantPool() is reachable: The declaring class of this element has been substituted, but this element is not present in the substitution class
2022-03-24T15:10:14.2127312Z To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
2022-03-24T15:10:14.2128517Z Detailed message:
2022-03-24T15:10:14.2171760Z Trace: 
2022-03-24T15:10:14.2172675Z 	at parsing java.lang.System$2.getConstantPool(System.java:2262)
2022-03-24T15:10:14.2173314Z Call path from entry point to java.lang.System$2.getConstantPool(Class): 
2022-03-24T15:10:14.2173740Z 	at java.lang.System$2.getConstantPool(System.java:2262)
2022-03-24T15:10:14.2174266Z 	at java.lang.reflect.RecordComponent.getAnnotatedType(RecordComponent.java:154)
2022-03-24T15:10:14.2174769Z 	at recordannotations.Main.main(Main.java:38)
2022-03-24T15:10:14.2175208Z 	at com.oracle.svm.core.JavaMainWrapper.runCore(JavaMainWrapper.java:150)
2022-03-24T15:10:14.2175769Z 	at com.oracle.svm.core.JavaMainWrapper.run(JavaMainWrapper.java:186)
2022-03-24T15:10:14.2176568Z 	at com.oracle.svm.core.code.IsolateEnterStub.JavaMainWrapper_run_5087f5482cc9a6abc971913ece43acb471d2631b(generated:0)
2022-03-24T15:10:14.2177001Z 
2022-03-24T15:10:14.2177259Z Error: Use -H:+ReportExceptionStackTraces to print stacktrace of underlying exception
```

**Comment by Karm** [comment]
@jerboaa  The CI failure from https://github.com/Karm/mandrel-integration-tests/pull/79 is tracked here.

**Comment by jerboaa** [comment]
@Karm I don't think `GraalVM 22.0.0.2-Final` has the fix, has it? Only `21.3.1.1-Final` has and `22.1` should have it.

**Comment by jerboaa** [comment]
> @jerboaa The CI failure from [Karm/mandrel-integration-tests#79](https://github.com/Karm/mandrel-integration-tests/pull/79) is tracked here.

Not sure what this means. https://github.com/jboss-container-images/openjdk/blob/develop/modules/prometheus/7/module.yaml#L26..L29 was a test on a GraalVM build that (knowingly) didn't have the fix.
